scripts/generate_site_assets.py

```
@@ -0,0 +1,156 @@
+    try:
+        exporter = ICalExporter(h_obj)
+        ics_path = year_dir / f"{filename_base}.ics"
+        exporter.save_ics(str(ics_path))
+
+        # Inline JSON generation
+        json_data = json.dumps(
+            [{"date": dt.isoformat(), "name": name} for dt, name in sorted(h_obj.items())],
+            ensure_ascii=False,
+        )
+        (year_dir / f"{filename_base}.json").write_text(json_data, encoding="utf-8", newline="\n")
+    except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)